pkg/csi/service/osutils/linux_os_utils.go

```diff
@@ -832,6 +849,138 @@ func (osUtils *OsUtils) GetDevMounts(ctx context.Context,
 	return devMnts, nil
 }
 
+// getMountInfo scans through /proc/self/mountinfo and
+// returns complete information for each entry in the file.
+func getMountInfo(ctx context.Context) ([]MountInfoDetail, error) {
+	infos := make([]MountInfoDetail, 0)
+	content, err := readProcsFile(procMountsPath)
+	if err != nil {
+		return infos, err
+	}
+	buffer := bytes.NewBuffer(content)
+	infos, err = readProcMountsFrom(ctx, buffer, true)
+	return infos, err
+}
+
+// readProcMountsFrom reads through the given file
+// and construct MountInfo struct objects.
+func readProcMountsFrom(ctx context.Context,
+	file io.Reader,
+	quick bool) ([]MountInfoDetail, error) {
+
+	var (
+		infos []MountInfoDetail
+		fscan = bufio.NewScanner(file)
+		cache = map[string]string{}
+	)
+
+	for fscan.Scan() {
+
+		// Read the next line of text and attempt to parse it into
+		// distinct, space-separated fields.
+		line := fscan.Text()
+		fields := strings.Fields(line)
+
+		// Remove the optional fields that should be ignored.
+		for {
+			val := fields[6]
```

Comment:
Can we make sure to do the size check if we absolutely require to keep this function in this PR?